frontend/components/ThemeProvider/ThemeProvider.tsx

```diff
@@ -0,0 +1,37 @@
+import React from "react";
+import { ThemeProvider } from "styled-components";
+import { getTheme, ITheme } from "./Theme";
+
+interface IThemeContext {
+  dark: boolean;
+  onToggle?: () => void;
+}
+
+const ThemeContext = React.createContext<IThemeContext>({
+  dark: false,
+});
+
+const useTheme = (): IThemeContext => React.useContext(ThemeContext);
+
+const ThemeContextProvider: React.FC = ({ children }) => {
+  const [dark, setDarkTheme] = React.useState<boolean>(false);
+  const theme: ITheme = dark ? getTheme("dark") : getTheme("light");
+
+  const handleToggle = (): void => {
+    setDarkTheme(!dark);
+  };
+  return (
+    <ThemeProvider theme={theme}>
```

Comment:
Additionally to changing to class for the 2 themes, this would be something like...

`theme={dark ? new Theme("dark").get() : new Theme("light").get}`

:P Not need all stuff above then XD

Reply:
sounds cool. May need your help. Will try it.